src/tools/rbd_wnbd/rbd_wnbd.cc

```diff
@@ -25,6 +25,11 @@
 #include "wnbd_handler.h"
 #include "rbd_wnbd.h"
 
+// LOCK_WRITE is also defined by objidl.h, we have to avoid
+// a collision.
+#undef LOCK_WRITE
```

Comment:
Can you elaborate on what you mean by a `objidl.h` wrapper?

I think `#define LOCK_WRITE 128` needs to happen after `objidl.h` is included in the translation unit so that the POSIX definition wins.  This is a POSIX code base with some Windows-specific bits, not the other way around, so unless a particular Windows constant is actually used preference should be given to POSIX definitions.  Perhaps including `objidl.h` separately before most other things are dragged in would do it?

Reply:
I was thinking about a wrapper that includes ``objidl.h`` and then restores the expected ``LOCK_WRITE`` value so that it could be included anywhere. Probably easier to just include ``objidl.h`` first.